plugins/plugin.go

```diff
@@ -45,6 +46,7 @@ type Plugin struct {
 	MetricsHost       string
 	Log               *logrus.Entry
 	ConnectorFunc     func([]producers.MetricsMessage, *cli.Context) error
+	Client            *http.Client
```

Comment:
you can make a single instance of http.Client and pass a reference to it to every function it needs.
`MetricsHost` can have unexported field `client`. But this is up to you i think it is more like a nit, not a deal breaker. If you feel like you need to export Client for now, it ok